client/ayon_core/tools/sceneinventory/model.py

```diff
@@ -129,29 +129,54 @@ def refresh(self, selected=None):
 
         self._clear_items()
 
-        items_by_repre_id = {}
+        items_by_repre_id = collections.defaultdict(list)
```

Comment:
This should be `items_by_project_and_repre_id` because `repre_id` may not be unique across projects. So two projects may use the same repre id, but would need different `items`.

Reply:
This may be better off as a nested instead:

```python
item_by_repre_id_by_project_id = defaultdict(lambda: defaultdict(set))
for container_item in container_items:
        project_name = (
            container_item.project_name or
            self._controller.get_current_project_name()
        )
        repre_id = container_item.representation_id
        item_by_repre_id_by_project_id[project_name][repre_id].add(container_item)
``` 
Maybe like this ☝️

Reply:
I find out that using this code would cause the scene inventory loading the invalid name and version(Entity n/a) It can contribute to this below https://github.com/ynput/ayon-core/pull/1005#discussion_r1843904610